kernel/kernel-api/src/test/scala/io/delta/kernel/internal/TableConfigSuite.scala

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright (2024) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel.internal
+
+import io.delta.kernel.data.{ColumnarBatch, ColumnVector, MapValue}
+import io.delta.kernel.internal.actions.{Format, Metadata}
+import io.delta.kernel.internal.util.VectorUtils
+import io.delta.kernel.internal.util.VectorUtils.{stringArrayValue, stringVector}
+import io.delta.kernel.test.{BaseMockJsonHandler, MockEngineUtils}
+import io.delta.kernel.types.{DataType, MapType, StringType, StructType}
+import org.scalatest.funsuite.AnyFunSuite
+
+import java.util.{Collections, Optional}
+import scala.collection.JavaConverters._
+
+class TableConfigSuite extends AnyFunSuite with MockEngineUtils {
+  def getEmptyMetadata: Metadata = {
+    new Metadata(
+      java.util.UUID.randomUUID().toString,
+      Optional.empty(),
+      Optional.empty(),
+      new Format(),
+      "",
+      null,
+      stringArrayValue(Collections.emptyList()),
+      Optional.empty(),
+      VectorUtils.stringStringMapValue(Collections.emptyMap())
+    )
+  }
+
+  test("Coordinated Commit Related Properties from Metadata") {
+    val expMap = Map("key1" -> "string_value", "key2Int" -> "2", "key3ComplexStr" -> "\"hello\"")
+    val engine1 = mockEngine(jsonHandler = new KeyValueJsonHandler(expMap))
+    val m1 = getEmptyMetadata.withNewConfiguration(
```

Comment:
you can actually make the `parseJSONKeyValueMap` method `protected` and write UTs just for that method. It avoids need for `Metadata` and simplifies the test code.